### PR TITLE
Improvement: create dialog composable

### LINKED ISSUE #145: Improvement: reduce duplication in Detach and Attach Dialogs

### PR DESCRIPTION
**Pull request by justinehell**
Resolve #145

### CODE REVIEW

File: src/composables/DetachDialog.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if (!isCurrentUser.value || (isCurrentUser.value && userStore.isAdmin)) {
+      reloadEvent();
+    }
+  }
+
+  /**
+   * Detach user, send event to reload users and close dialog.
+   * @returns {Promise<void>} Promise with nothing on success.
+   */
+  async function onSubmit() {
```
Comment (Zorin95670) [inline src/composables/DetachDialog.js:96]: rename to detach

File: src/composables/DetachDialog.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    return ReloadRolesEvent.next();
+  }
+
+  /**
+   * Invoke the appropriate method from GroupService or RoleService.
+   * @returns {Promise<void>} Promise with nothing on success.
+   */
+  async function detach() {
```
Comment (Zorin95670) [inline src/composables/DetachDialog.js:63]: rename to executeDetach or another

File: src/composables/AttachDialog.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    return ReloadPermissionsEvent.next();
+  }
+
+  /**
+   * Invoke the appropriate method from GroupService or RoleService.
+   * @param {string} selectedId - Selected id of the item to be attached.
+   * @returns {Promise<void>} Promise with nothing on success.
+   */
+  async function attach(selectedId) {
```
Comment (Zorin95670) [inline src/composables/AttachDialog.js:75]: same here rename to associate or invokeAttach

File: src/composables/AttachDialog.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    });
+
+    selected.value = failedRequestObjects;
+  }
+
+  /**
+   * Attach one or more groups to a user.
+   * @returns {Promise<void>} Promise with nothing on success.
+   */
+  async function onSubmit() {
```
Comment (Zorin95670) [inline src/composables/AttachDialog.js:121]: same here rename to attach

File: src/composables/DetachDialog.js (hunk 1 above)
Comment (justinehell) [reply src/composables/DetachDialog.js]: done

File: src/composables/DetachDialog.js (hunk 2 above)
Comment (justinehell) [reply src/composables/DetachDialog.js]: done

File: src/composables/AttachDialog.js (hunk 3 above)
Comment (justinehell) [reply src/composables/AttachDialog.js]: done

File: src/composables/AttachDialog.js (hunk 4 above)
Comment (justinehell) [reply src/composables/AttachDialog.js]: done

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**